Add aws_elb resource

sundowndev-snyk opened the pull request:
| Q                 | A
| ----------------- | ---
| 🐛 Bug fix?       | no
| 🚀 New feature?   | yes
| ⚠ Deprecations?   | no
| ❌ BC Break       | no
| 🔗 Related issues | #919 
| ❓ Documentation  | https://github.com/snyk/driftctl-docs/pull/236 <!-- does this require documentation update ? -->

## Description

This PR add support for the `aws_elb` resource.

sundowndev-snyk committed:
feat: add aws_elb resource

wbeuil commented on pkg/resource/aws/classic_loadbalancer.go:
```diff
@@ -0,0 +1,3 @@
+package aws
```
I would have called this file just like the Terraform resource type `aws_elb.go` (be aware of the AWS prefix) and change the test file as well. 

But the naming is good `AwsClassicLoadBalancerResourceType`

wbeuil commented on pkg/remote/aws/repository/elb_repository.go:
```diff
@@ -0,0 +1,46 @@
+func (r *elbRepository) ListAllLoadBalancers() ([]*elb.LoadBalancerDescription, error) {
+	cacheKey := "elbListAllLoadBalancers"
+	v := r.cache.GetAndLock(cacheKey)
+	defer r.cache.Unlock(cacheKey)
```
It's not really mandatory right now to do this (GetAndLock and defer the Unlock) since we're not using this method in any other enumerator.

wbeuil commented on pkg/remote/aws/repository/elb_repository.go:
```diff
@@ -0,0 +1,46 @@
+}
+
+func (r *elbRepository) ListAllLoadBalancers() ([]*elb.LoadBalancerDescription, error) {
+	cacheKey := "elbListAllLoadBalancers"
```
We're gonna have a problem here since this is the same `cacheKey` than the one in the ELBv2 resource.

Let's change the one in the v2 resource to `elbv2ListAllLoadBalancers` to be inlined with the rest.

sundowndev-snyk replied:
Very good catch!!

wbeuil commented on pkg/remote/aws_elb_scanner_test.go:
```diff
@@ -0,0 +1,104 @@
+			assertExpected: func(t *testing.T, got []*resource.Resource) {
+				assert.Len(t, got, 0)
+			},
+		},
```
Can you add one with a dummy error

wbeuil reviewed (changes requested):
Small comments but the one with the cacheKey is really important!

sundowndev-snyk committed:
refactor: aws_elb resource

Fix cache keys in ELB repository, fix naming and add a test case in scanner tests.

sundowndev-snyk committed:
test: elbv2 repository

wbeuil reviewed (approved):
LGTM
